### PR TITLE
tikv config: Add in-memory lock tikv config

### PR DESCRIPTION
**Pull request by cfzjywxk**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -2306,6 +2306,18 @@ Raft Engine 相关的配置项。
 + 默认值：true
 + 注意：`in-memory` 仅在 `pipelined` 为 true 时生效。
 
+### `in-memory-peer-size-limit` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
++ 控制单 region 内存悲观锁内存使用上限。当使用超过限制时，悲观锁将 fallback 到持久化方式写入。
```
Comment (ekexium) [inline tikv-configuration-file.md:2311]: fallback 在中文文档里是一个常用用法吗？还是「回退」？

**Review by ekexium — APPROVED** [review]

File: tikv-configuration-file.md (hunk 1 above)
Comment (Oreoxmt) [inline tikv-configuration-file.md:2311]:
```suggestion
+ 控制单个 Region [内存悲观锁](/pessimistic-transaction.md#内存悲观锁)的内存使用上限。超过此限制时，悲观锁将回退到持久化方式写入。
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -2306,6 +2306,18 @@ Raft Engine 相关的配置项。
...
++ 默认值：512KiB
++ 单位：KiB|MiB|GiB
+
+### `in-memory-instance-size-limit` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
++ 控制单 tikv 实例内存悲观锁使用总上限。当使用超过限制时，悲观锁将 fallback 到持久化方式写入。
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:2317]:
```suggestion
+ 控制单个 TiKV 实例[内存悲观锁](/pessimistic-transaction.md#内存悲观锁)的总内存使用上限。超过此限制时，悲观锁将回退到持久化方式写入。
```

File: dynamic-config.md [hunk 3]
```diff
@@ -172,6 +172,8 @@ show warnings;
 | pessimistic-txn.wake-up-delay-duration | 悲观事务被重新唤醒的时间 |
 | pessimistic-txn.pipelined | 是否开启流水线式加悲观锁流程 |
 | pessimistic-txn.in-memory | 是否开启内存悲观锁功能 |
+| pessimistic-txn.in-memory-peer-size-limit | 控制单 region 内存悲观锁内存使用上限 |
+| pessimistic-txn.in-memory-instance-size-limit | 控制单 TiKV 实例内存悲观锁内存使用上限 |
```
Comment (Oreoxmt) [inline dynamic-config.md:176]:
```suggestion
| pessimistic-txn.in-memory-peer-size-limit | 控制单个 Region 内存悲观锁的内存使用上限 |
| pessimistic-txn.in-memory-instance-size-limit | 控制单个 TiKV 实例内存悲观锁的内存使用上限 |
```

**Review by Oreoxmt — APPROVED** [review]

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -2306,6 +2306,18 @@ Raft Engine 相关的配置项。
 + 默认值：true
 + 注意：`in-memory` 仅在 `pipelined` 为 true 时生效。
 
+### `in-memory-peer-size-limit` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
++ 控制单个 Region [内存悲观锁](/pessimistic-transaction.md#内存悲观锁)的内存使用上限。超过此限制时，悲观锁将回退到持久化方式写入。
```
Comment (qiancai) [inline tikv-configuration-file.md:2311]: 这里的 “回退到持久化方式写入” 就是指 回退为使用 [pipelined 加锁流程](https://docs.pingcap.com/zh/tidb/dev/pessimistic-transaction#pipelined-%E5%8A%A0%E9%94%81%E6%B5%81%E7%A8%8B)吗

Comment (qiancai) [inline tikv-configuration-file.md:2311]:
```suggestion
+ 控制单个 Region [内存悲观锁](/pessimistic-transaction.md#内存悲观锁)的内存使用上限。超过此限制时，悲观锁将回退到持久化方式写入磁盘。
```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -2306,6 +2306,18 @@ Raft Engine 相关的配置项。
...
++ 默认值：512KiB
++ 单位：KiB|MiB|GiB
+
+### `in-memory-instance-size-limit` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
++ 控制单个 TiKV 实例[内存悲观锁](/pessimistic-transaction.md#内存悲观锁)的内存使用上限。超过此限制时，悲观锁将回退到持久化方式写入。
```
Comment (qiancai) [inline tikv-configuration-file.md:2317]:
```suggestion
+ 控制单个 TiKV 实例[内存悲观锁](/pessimistic-transaction.md#内存悲观锁)的内存使用上限。超过此限制时，悲观锁将回退到持久化方式写入磁盘。
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -2306,6 +2306,18 @@ Raft Engine 相关的配置项。
 + 默认值：true
 + 注意：`in-memory` 仅在 `pipelined` 为 true 时生效。
```
[the remark this answers is not in the document]
Comment (Oreoxmt) [reply tikv-configuration-file.md:2308]: @cfzjywxk PTAL

[the remark this answers is not in the document]
Comment (cfzjywxk) [reply tikv-configuration-file.md:2312]: Added.

File: pessimistic-transaction.md [hunk 7]
```diff
@@ -186,7 +186,7 @@ set config tikv pessimistic-txn.pipelined='false';
 
 TiKV 在 v6.0.0 中引入了内存悲观锁功能。开启内存悲观锁功能后，悲观锁通常只会被存储在 Region leader 的内存中，而不会将锁持久化到磁盘，也不会通过 Raft 协议将锁同步到其他副本，因此可以大大降低悲观事务加锁的开销，提升悲观事务的吞吐并降低延迟。
 
-当内存悲观锁占用的内存达到 Region 或节点的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
+当内存悲观锁占用的内存达到 [Region](/tikv-configuration-file.md#in-memory-peer-size-limit-span-classversion-mark从-v840-版本开始引入span) 或[节点](/tikv-configuration-file.md#in-memory-instance-size-limit-span-classversion-mark从-v840-版本开始引入span)的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
```
Comment (Oreoxmt) [inline pessimistic-transaction.md:189]:
```suggestion
当内存悲观锁占用的内存达到 [Region](/tikv-configuration-file.md#in-memory-peer-size-limit-从-v840-版本开始引入) 或[节点](/tikv-configuration-file.md#in-memory-instance-size-limit-从-v840-版本开始引入)的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
```

**Review by Oreoxmt — APPROVED** [review]

File: pessimistic-transaction.md [hunk 8]
```diff
@@ -186,7 +186,7 @@ set config tikv pessimistic-txn.pipelined='false';
 
 TiKV 在 v6.0.0 中引入了内存悲观锁功能。开启内存悲观锁功能后，悲观锁通常只会被存储在 Region leader 的内存中，而不会将锁持久化到磁盘，也不会通过 Raft 协议将锁同步到其他副本，因此可以大大降低悲观事务加锁的开销，提升悲观事务的吞吐并降低延迟。
 
-当内存悲观锁占用的内存达到 Region 或节点的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
+当内存悲观锁占用的内存达到 [Region](/tikv-configuration-file.md#in-memory-peer-size-limit-从-v840-版本开始引入) 或[节点](/tikv-configuration-file.md#in-memory-instance-size-limit-从-v840-版本开始引入)的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
```
Comment (qiancai) [inline pessimistic-transaction.md:189]:
```suggestion
当内存悲观锁占用的内存达到 [Region](/tikv-configuration-file.md#in-memory-peer-size-limit-从-v840-版本开始引入) 或[ TiKV 节点](/tikv-configuration-file.md#in-memory-instance-size-limit-从-v840-版本开始引入)的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: pessimistic-transaction.md [hunk 9]
```diff
@@ -186,7 +186,7 @@ set config tikv pessimistic-txn.pipelined='false';
 
 TiKV 在 v6.0.0 中引入了内存悲观锁功能。开启内存悲观锁功能后，悲观锁通常只会被存储在 Region leader 的内存中，而不会将锁持久化到磁盘，也不会通过 Raft 协议将锁同步到其他副本，因此可以大大降低悲观事务加锁的开销，提升悲观事务的吞吐并降低延迟。
 
-当内存悲观锁占用的内存达到 Region 或节点的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
+当内存悲观锁占用的内存达到 [Region](/tikv-configuration-file.md#in-memory-peer-size-limit-从-v840-版本开始引入) 或[ TiKV 节点](/tikv-configuration-file.md#in-memory-instance-size-limit-从-v840-版本开始引入)的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
```
Comment (Oreoxmt) [inline pessimistic-transaction.md:189]:
```suggestion
当内存悲观锁占用的内存达到 [Region](/tikv-configuration-file.md#in-memory-peer-size-limit-从-v840-版本开始引入) 或 [TiKV 节点](/tikv-configuration-file.md#in-memory-instance-size-limit-从-v840-版本开始引入)的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
```

**Pull request merged**